cdap-securestore-ext-cloudkms/src/main/java/co/cask/cdap/securestore/spi/CloudSecretManager.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.securestore.spi;
+
+import co.cask.cdap.securestore.spi.secret.Secret;
+import co.cask.cdap.securestore.spi.secret.SecretMetadata;
+import com.google.common.collect.ImmutableList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+/**
+ * A {@link SecretManager} which manages sensitive data using Google Cloud KMS.
+ */
+public class CloudSecretManager implements SecretManager {
+  private static final Logger LOG = LoggerFactory.getLogger(CloudSecretManager.class);
+  private static final String CLOUD_KMS_NAME = "cloudkms";
+  private SecretStore store;
+  private CloudKMSClient client;
+  private SecretEncoderDecoder encoderDecoder;
+
+  @Override
+  public String getName() {
+    return CLOUD_KMS_NAME;
+  }
+
+  @Override
+  public void initialize(SecretManagerContext context) throws IOException {
+    this.store = context.getSecretStore();
+    this.client = new CloudKMSClient();
+    this.client.createKeyRing();
+    this.encoderDecoder = new SecretEncoderDecoder();
+  }
+
+  @Override
+  public void store(String namespace, Secret secret) throws IOException {
+    // creates new crypto key for every namespace
+    client.createCryptoKey(namespace);
```

Comment:
Seems like the namespace will be used directly as the crytoKeyId? I think it is better to have it prefixed